pkg/cloudprovider/aws/instance.go

```diff
@@ -240,8 +248,9 @@ func (p *InstanceProvider) instanceToNode(instance *ec2.Instance, instanceTypes
 						v1.ResourceMemory: *instanceType.Memory(),
 					},
 					NodeInfo: v1.NodeSystemInfo{
-						Architecture:    v1alpha1.AWSToKubeArchitectures[aws.StringValue(instance.Architecture)],
-						OSImage:         aws.StringValue(instance.ImageId),
+						Architecture: v1alpha1.AWSToKubeArchitectures[aws.StringValue(instance.Architecture)],
+						OSImage:      aws.StringValue(instance.ImageId),
+						//TODO: stop hard-coding this
```

Comment:
I think you can drop this TODO.

Reply:
I can drop it, I just wanted to understand why. Is this a stylistic thing with TODOs? Do you think we'll never change this? If so, why?

Reply:
Generally we try to avoid them, they just tend to pile up and aren't really a great way to track future work (compared to filing an issue).

Reply:
Again: totally ok to abide by your conventions, but just wondering: why not just add a TODO and link to the issue? I offered to create an issue but Ellis said it wasn't necessary, which is why I didn't do that here. I generally tell people _to_ add TODOs and link to the issues created so that the spot in the code is easier to find by whoever is working the issue and also so that passerbys in the code base understand that it's a known issue.

Reply:
I don't think we really know what we want to do with OS at this point. In general, the custom LT story isn't fleshed out. Instead of guessing where we might need lines of code, I think it's just as good to just figure that out when we're fleshing out the OS story. It helps avoid leaving spurious TODOs around the codebase that may or may not ever get addressed.